tests/admin/controller/test_metadata_services.py

```diff
@@ -193,14 +214,129 @@ def test_metadata_services_post_edit(self):
                 (ExternalIntegration.PASSWORD, "pass"),
                 ("libraries", json.dumps([{"short_name": "L2"}])),
             ])
-            response = self.manager.admin_metadata_services_controller.process_metadata_services()
+            response = controller.process_post()
             eq_(response.status_code, 200)
 
-        eq_(novelist_service.id, int(response.response[0]))
-        eq_(ExternalIntegration.NOVELIST, novelist_service.protocol)
-        eq_("user", novelist_service.username)
-        eq_("pass", novelist_service.password)
-        eq_([l2], novelist_service.libraries)
+    def test_metadata_services_post_calls_register_with_metadata_wrangler(self):
```

Comment:
This test doesn't prove that the problem has been fixed -- register_with_metadata_wrangler was being called before, but it was exiting out before the crash.